sup/include/sup/gpu.hpp

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <arbor/version.hpp>
+
+#ifdef ARB_MPI_ENABLED
+#include <mpi.h>
+#endif
+
+namespace sup {
+
+int find_gpu();
+
+#ifdef ARB_MPI_ENABLED
+int find_gpu(MPI_Comm comm);
+#endif
+
+} // namespace sup
+
```

Comment:
As above, `find_gpu(MPI_Comm)` does not require that `arbor` was built with MPI, but that MPI support is available at configure time. The MPI code can live in a different header+source that just includes `mpi.h`, but which is only conditionally included in the `sup` library based on the configure-time `ARB_WITH_MPI`.

The `find_gpu` names for these functions are too generic and don't really reflect what they do, which itself is very different in `find_gpu()` and `find_gpu(MPI_Comm)`.

The non-distributed version just checks to see if any GPU is visible, and so should have a corresponding name and return true or false. Alternately, if the interface doesn't change, it should have a name like `sup::default_gpu()` or similar.

The distributed one does this negotiation between ranks, and perhaps should be called something like `sup::find_private_gpu_id(MPI_Comm)` or something else along these lines.

Reply:
I would be for keeping the interfaces, and renaming them. `sup::default_gpu` and `sup::find_private_gpu` are quite good names for that purpose!